src/routes/(console)/organization-[organization]/usage/[[invoice]]/+page.svelte

```diff
@@ -21,32 +21,39 @@
     import { formatCurrency, formatNumberWithCommas } from '$lib/helpers/numbers';
     import { Icon, Layout, Link, Tooltip, Typography } from '@appwrite.io/pink-svelte';
     import { IconChartSquareBar, IconInfo } from '@appwrite.io/pink-icons-svelte';
+    import { onMount } from 'svelte';
+    import type { UsageProjectInfo } from '../../store';
 
     export let data;
 
     const tier = data?.plan
         ? (data.plan.$id as Tier)
         : (data?.currentInvoice?.plan ?? $organization?.billingPlan);
+
     const plan = data?.plan ?? undefined;
 
     $: projects = (data.organizationUsage as OrganizationUsage).projects;
 
+    let usageProjects: { [key: string]: UsageProjectInfo } = {};
+
     $: legendData = [
         {
             name: 'Reads',
             value: data.organizationUsage.databasesReads.reduce(
-                (sum, singleDay) => sum + singleDay.value,
+                (sum: number, singleDay: { date: string; value: number }) => sum + singleDay.value,
                 0
             )
         },
         {
             name: 'Writes',
             value: data.organizationUsage.databasesWrites.reduce(
-                (sum, singleDay) => sum + singleDay.value,
+                (sum: number, singleDay: { date: string; value: number }) => sum + singleDay.value,
                 0
             )
         }
     ];
+
+    onMount(async () => (usageProjects = await data.projectsPromise));
```

Comment:
why dont we pass the promise and await where we actually need it? 👀

Reply:
Its used on multiple `ProjectBreakdown` components based on the type of service. So it makes sense to load all of them here.

Reply:
Also it adds a bit of redundant logic to maintain a promise and a resolved var with value under `ProjectBreakdown`. Doesn't feel like the component's responsibility.